### DISCUSSION TITLE
act要素

### DISCUSSION THREAD

**Discussion by scarsty** [scarsty/kys-cpp#46] [Ideas]
八方向像素级移动，非之前不连续移动
轻攻击，重攻击，闪身
攻击硬直
重击时震动
打中时画面停，效果，声音
受伤硬直

**Comment by scarsty** [comment]
打断 霸体等

**Comment by scarsty** [comment]
方向需改成向量 以两个值保存 用于计算攻击位置
击退需根据敌方攻击方向
攻击效果用deque或list保存 每帧自动清除放完的

**Comment by scarsty** [comment]
拳法无特殊 剑法多点低攻击 刀法打退 特殊拉近
武功就是加特效 不影响攻击值
攻击力影响打断等
轻攻击牵制 重击带特效
单通

**Comment by chopstickskz** [comment]


**Reply by scarsty** [reply]
damage over time  百度的 哈哈哈

**Comment by scarsty** [comment]
分段蓄力 完美释放等
蓄力足才有效果

**Comment by scarsty** [comment]
自动瞄准 攻击方向 模仿维吉尔疾走居合等

**Comment by scarsty** [comment]
四种攻击 轻攻击消耗低 可以连发 重攻击蓄力可以一次打出多发 可能有追踪效果 远程攻击打直线 自动瞄准 一次一发 闪身攻击随闪身发动 有多发 每种攻击力略有区别 体力回复加快

**Comment by scarsty** [comment]
武功可以自有特效

**Comment by scarsty** [comment]
本垃圾先瞎说几个，抛砖引玉
紫霞：体力恢复加倍
小无: 敌方僵直时长+25%
十八泥偶: 每10秒钟恢复10%内力值
神照经: 生命低于40%霸体10秒钟
易筋经: +10 -10 最终伤害
洗髓经:  每10秒钟恢复5%生命值
龙象: 30%免疫僵直打断抗性
太玄: 闪身时给敌方造成一次攻击伤害
乾坤大挪移: 攻击者受到5点伤害
葵花宝典: 僵直状态受到伤害时随机闪身
九阴:  每秒钟最终伤害+1，直到25秒上限
九阳：内力消耗-50%

**Reply by scarsty** [reply]
已完成紫霞，小无，十八，乾坤

**Reply by scarsty** [reply]
洗髓

**Comment by scarsty** [comment]
因原版的武功设定过于简单 可能改用苍龙逐日为基础

**Comment by chopstickskz** [comment]
核心
参照哈迪斯、暖雪、以撒等相关游戏，核心玩法是build搭配及随机性。
山寨到金群可进行一下修改
1.武功
  1）摒弃原金群武功等级修炼方式等无用功能，武功拾取及习得。无武功等级概念.
  2）每种武学设定3类用于区分. 普通武功 江湖武功 绝学。并且设置2种属性，攻击属性/防御属性
       (1)普通/江湖：点攻击,低耗内.无附带异常效果.基础威力低，无CD.
           设定参照：参照普通攻击
       (2)江湖/门派：点、线攻击.或可最多同时攻击X人，或可依据武功动画弹道进行多段打击，附带异常状态，触发几率中。基础威力中。短CD。
           设定参照：参照特殊攻击
       (3)绝学：攻击范围根据武功特性，或可最多同时攻击X人，或可依据武功动画弹道进行多段打击。附带异常状态，触发几率高
           设定参照：参照无双（爆发技）
       (4)防招：任何武学均可装备之守招包含当前使用武学。引用武学设定中的防御属性。
            设定参照：网金OL的格挡武学，人在江湖的守招系统。
  3）武功威力/防招：武学基础威力、4维系数及心法、门派、其他武学等加成影响.
  4）异常状态/抗性：晕、流血、减速(断筋)、中毒、内伤、致盲。
  5）攻击频率：依据门派特性、内功心法属性、4围系数等 提升普通武功攻击速度、减少江湖武功、绝学的武功CD
  6）轻功：
          (1)使用方式：单独按键，可触发原地/移动闪身 躲避攻击动画，次数有上限，使用后转CD充填次数。且触发闪身后增加BUFF效果。
          (2)根据不同的轻功设定使用上限，充填CD及BUFF效果。4种左右即可，如：使用轻功后增加下一次攻击伤害、清除武功技能CD。下一次被攻击降低伤害等。
          (3)轻功属性加成充填CD速度。
       或取消闪身效果、改为被动触发MISS几率 触发后增加BUFF效果

  7）内功：区分学习效果及运功效果及三老师内功意见稿 如图(人在江湖牛逼)
![image](https://user-images.githubusercontent.com/99104820/154253738-353bc875-a844-4682-8a46-263d1b2d3015.png)
  8）武学搭配：
          (1).武学、内功与门派特性搭配：及武学受到内功及门派特性效果影响
               如：绝学降龙十八掌，主运内功九阴、门派丐帮 = 门派加成+内功加成+降龙威力 （文字特效?或直接加成伤害，随意）
          (2).武学与武器搭配：玄铁剑法、玄铁剑，打狗棒法、打狗棒等搭配加成。
          (3).武学与异常状态搭配：例：玄铁剑法增加x几率造成对方内伤，对已经内伤的目标造成的伤害增加15%  等等
  9）武学特性：击退(增加目标技能CD)、连击、暴击等等等等等(编不下去了 艹)
 10）武学限制：分别限制武学、轻功、内功习得数量。
2.门派
  1）每个门派设定独有的门派特性. 如武当，门派主修拳掌、剑法，增加兵器值，武功获得概率。
  2）门派武功加成.加成太极拳剑、太极神功。
  3）提高本门绝学获得几率.
3.装备
  不建议套装体系.装备除增加基础数值外，可增加部分武功威力或系数武功威力及异常状态触发几率
4.商店
  增加可出售的武学(武功、轻功、内功)、装备。药品等
5.增加随机性、武学、装备随机掉落，击杀银两奖励等！
6.战斗效果：素材过于垃圾动画效果是否可改为弹幕式？即多段伤害。连击、江湖/门派武学、绝学可打出多弹幕效果。
7.框架改为单线性模式。略
